Makefile

```
@@ -928,13 +927,16 @@ endif
 
 # Check that the stack limit is reasonable (Unix-only)
 .PHONY: checkstack
-checkstack:
 ifeq "$(UNIX_OR_WIN32)" "unix"
-	if $(MKEXE) $(OUTPUTEXE)tools/checkstack$(EXE) tools/checkstack.c; \
-	  then tools/checkstack$(EXE); \
-	fi
-	rm -f tools/checkstack$(EXE)
+checkstack: tools/checkstack$(EXE)
+	$<
+	rm -f $<
+
+tools/checkstack$(EXE): tools/checkstack.$(O)
+	$(MKEXE) $(OUTPUTEXE)$@ $<
+	rm $^
```

```suggestion
checkstack: tools/checkstack$(EXE)
	$<

.INTERMEDIATE: tools/checkstack$(EXE) tools/checkstacxk.$(O)
tools/checkstack$(EXE): tools/checkstack.$(O)
	$(MKEXE) $(OUTPUTEXE)$@ $<
```

This has the nice benefit that GNU make automatically removes the executable and object even if running the program fails.